discovery-provider/src/solana/anchor_program_indexer.py

```diff
@@ -92,8 +89,9 @@ async def parse_tx(self, tx_sig):
         # Append each parsed transaction to parsed metadata
         tx_instructions = []
         for instruction in tx.instructions:
-            parsed_instr = parse_instruction(instruction, self._instruction_coder)
-            tx_instructions.append(parsed_instr)
+            parsed_instr = self.anchor_parser.parse_instruction(instruction)
+            if self.is_valid_instruction(parsed_instr):
```

Comment:
so one clarification - will this work when parsing the `CreateUser` call with 2 instructions? or do we end up skipping the 0th index tx

Reply:
bc we have a tx with [secpInstruction, createUserInstruction]

Reply:
actually with current structure this will be fine since the instruction name is unknown and we don't have a case for that in is_valid_instruction we should return true. does that sound right?

Reply:
yup that's what i was thinking